### PR TITLE
Merge some code to prepare for new authorization system

### PR DESCRIPTION
**Pull request by jeremyevans**
This still uses the old authorization system, but merges changes required by the new authorization system that will not conflict.  This will allow breaking the switchover into smaller chunks.

### CODE REVIEW

File: model/project.rb [hunk 1]
```diff
@@ -70,6 +72,9 @@ def soft_delete
     DB.transaction do
       access_tags_dataset.destroy
       access_policies_dataset.destroy
+      access_control_entries_dataset.destroy
+      DB[:applied_subject_tag].where(tag_id: subject_tags_dataset.select(:id)).delete
```
Comment (enescakir) [inline model/project.rb:76]: Should we delete action and object tags too?

File: model/project.rb [hunk 2]
```diff
@@ -5,6 +5,8 @@
 class Project < Sequel::Model
   one_to_many :access_tags
   one_to_many :access_policies
+  one_to_many :access_control_entries
+  one_to_many :subject_tags
```
Comment (enescakir) [inline model/project.rb:9]: We might include the commit that adds also `object_tags`, `action_tags` to this PR too.

**Review by enescakir — APPROVED** [review]

File: model/project.rb (hunk 1 above)
Comment (jeremyevans) [reply model/project.rb]: Yes, that was an oversight, and looks like it wasn't in the main PR.  I updated this PR to include it, and will rebase the main PR on this PR.

File: model/project.rb (hunk 2 above)
Comment (jeremyevans) [reply model/project.rb]: That makes sense.  I updated the last commit in this PR to do that.

**Pull request merged**